sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -181,11 +182,7 @@ abstract class AggregationIterator(
         // Process all expression-based aggregate functions.
         updateProjection.target(currentBuffer)(joinedRow(currentBuffer, row))
         // Process all imperative aggregate functions.
-        var i = 0
-        while (i < updateFunctions.length) {
-          updateFunctions(i)(currentBuffer, row)
-          i += 1
-        }
+        updateFunctions.foreach(updateFunction => updateFunction(currentBuffer, row))
```

Comment:
For [1], it's written as below and it seems including functional transformations
> Use while loops instead of for loops or functional transformations (e.g. `map`, `foreach`).

Then, could you then confirm the bytecode is virtually the same or more efficient? My point was that I think we need a clear reason in this case.